blockprotocol/blockprotocol
# Allow direct links to blocks & schemas of a user

## What this Does

The user profile page on the BlockProtocol site currently has tabs to navigate between a user's profile, their blocks, and available schemas. However, one presently cannot link to the blocks or schemas tabs directly. This PR creates those pages by refactoring existing code, allowing users to be able to share links to their blocks or schemas pages freely.

## What Does this Change

- [Add slugs to tabs](https://github.com/blockprotocol/blockprotocol/pull/220/files#diff-ea85b256b8482c1d52bf722baf3a828dbf2918d105c4400ec1b19a8607776851R8) for use during routing.
- [Create a `UserPageComponent`](https://github.com/blockprotocol/blockprotocol/pull/220/files#diff-34bc1d1615cd3a42118e7265c1dfc012e8a76c4d77d0529688c4ff95fcb1dfe3R1) for reuse across all user pages.
- Create seperate pages for blocks and schemas with the default tab set [to the respective page](https://github.com/blockprotocol/blockprotocol/pull/220/files#diff-a703eeb67bf29ca9dfa259afcba3e877663639b2209765b08f40765a49e7437fR62).

## How to Test this

- Go to a user's profile
- Switch between various tabs and look at the URL indicator
- Try to go to a specific URL directly

![chrome_J0eWKUP4M0](https://user-images.githubusercontent.com/22196279/154047329-022052e8-8f3a-4374-ac13-cc40e6f49df3.gif)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201768159094198
